packages/gatsby-plugin-catch-links/src/catch-links.js

```diff
@@ -75,6 +84,14 @@ module.exports = function(root, cb) {
 
     ev.preventDefault()
 
+
+    var anchoreUrl = new URL(anchor.getAttribute(`href`))
+
+    if (checkSameOriginWithoutProtocol(window.location.origin, anchoreUrl.origin)) {
+      cb(anchoreUrl.pathname)
```

Comment:
I think this will strip `#` and `?` parts from link? Maybe this should be ``cb(`${anchoreUrl.pathname}${anchoreUrl.search}${anchoreUrl.hash}`)``?

Reply:
@pieh Great catch! 🔥 
I'll make this change.

Reply:
Done. 👍